ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerSessionConfiguration.scala

```diff
@@ -7,13 +7,14 @@ import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantSessio
 import com.daml.ledger.api.tls.TlsConfiguration
 
 private[testtool] final case class LedgerSessionConfiguration(
-    participants: Vector[(String, Int)],
+    private val participantAddresses: Vector[(String, Int)],
+    private val ssl: Option[TlsConfiguration],
+    private val partyAllocation: PartyAllocationConfiguration,
     shuffleParticipants: Boolean,
-    ssl: Option[TlsConfiguration],
-    partyAllocation: PartyAllocationConfiguration,
 ) {
-  def forParticipant(hostAndPort: (String, Int)): ParticipantSessionConfiguration = {
-    val (host, port) = hostAndPort
-    ParticipantSessionConfiguration(host, port, ssl, partyAllocation)
-  }
+  lazy val participants: Vector[ParticipantSessionConfiguration] =
```

Comment:
If the only use case for this class is to produce `ParticipantSessionConfiguration`s, laziness is likely to not be useful.

Reply:
Removed the laziness. I think it was useful about half-way in, but not any more.